src/main/java/net/logstash/logback/stacktrace/StackHasher.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.logstash.logback.stacktrace;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/**
+ * Utility class that generates a hash for any Java {@link Throwable error} stack
+ *
+ * @author Pierre Smeyers
+ */
+public class StackHasher {
+
+    private final StackElementFilter filter;
+
+    /**
+     * Constructs a {@link StackHasher} with the given filter
+     *
+     * @param filter filter
+     */
+    public StackHasher(StackElementFilter filter) {
+        this.filter = filter;
+    }
+
+    /**
+     * Constructs a {@link StackHasher} using {@link StackElementFilter#any()} filter
+     */
+    public StackHasher() {
+        this(StackElementFilter.any());
+    }
+
+    /**
+     * Generates a Hexadecimal hash for the given error stack
+     * <p>
+     * Two errors with the same stack hash are most probably same errors
+     */
+    public String hexHash(Throwable error) {
+        return hexHashes(error).peek();
+    }
+
+    /**
+     * Generates and returns Hexadecimal hashes for the error stack and each ancestor {@link Throwable#getCause() cause}
+     * <p>
+     * The first queue element is the stack hash for the topmost error, the next one (if any) is it's direct
+     * {@link Throwable#getCause() cause} hash, and so on...
+     */
+    public Deque<String> hexHashes(Throwable error) {
+        Deque<String> hexHashes = new ArrayDeque<String>();
+        hash(error, hexHashes);
+        return hexHashes;
+    }
+
+    /**
+     * Generates a hash (int) of the given error stack
+     * <p>
+     * Two errors with the same stack hash are most probably same errors
+     */
+    int hash(Throwable error, Deque<String> hexHashes) {
```

Comment:
Do you want to consider the throwable's message in the hash?

Seems like throwables with different messages should generate different hashes, no?